lib/evse_security/evse_security.cpp

```diff
@@ -648,6 +658,17 @@ std::string EvseSecurity::get_verify_file(CaCertificateType certificate_type) {
     EVLOG_debug << "Requesting certificate file: [" << conversions::ca_certificate_type_to_string(certificate_type)
                 << "] file:" << verify_file.get_path();
 
+    // If we are using a directory, search for the first valid root file
+    if (verify_file.is_using_directory()) {
+        auto& hierarchy = verify_file.get_certficate_hierarchy();
+
+        // Get all roots and search for a valid self-signed
+        for (auto& root : hierarchy.get_hierarchy()) {
+            if (root.certificate.is_selfsigned() && root.certificate.is_valid())
+                return root.certificate.get_file().value_or("");
+        }
+    }
```

Comment:
Note: In case there are multiple roots, this interface is probably not adequate. We would need to be able to return a list of certificates somehow.